### PR TITLE
OMID-256 Bump hbase and other dependencies to latest version

### PR DESCRIPTION
**Pull request by NihalJain**
- Bump hbase.version to 2.4.17
- Bump log4j2.version to 2.21.0
- Bump junit.version to 4.13.2
- Bump commons-lang3.version to 3.13.0

### CODE REVIEW

**Comment by NihalJain** [comment]
Ran following command with a consolidated change of 4 PRs locally:
mvn clean verify site -Dspotbugs.site -Dhbase.version=2.5.5-hadoop3 -Dsurefire.rerunFailingTestsCount=3 
PR list:
- https://github.com/apache/phoenix-omid/pull/143
- https://github.com/apache/phoenix-omid/pull/144
- https://github.com/apache/phoenix-omid/pull/145
- https://github.com/apache/phoenix-omid/pull/146

BUILD was SUCCESS!

**Review by gjacoby126 — APPROVED** [review]
+1

**Review by stoty — CHANGES_REQUESTED** [review]
Please list the changes in the comit message.

File: pom.xml [hunk 1]
```diff
@@ -152,21 +152,21 @@
...
+        <log4j2.version>2.21.0</log4j2.version>
         <netty4.version>4.1.86.Final</netty4.version>
         <!-- com.google repo will be used except on Aarch64 platform. -->
         <protobuf.group>com.google.protobuf</protobuf.group>
         <protobuf-java.version>2.5.0</protobuf-java.version>
         <protoc.version>2.5.0</protoc.version>
         <protobuf.plugin.version>0.6.1</protobuf.plugin.version>
-        <os.plugin.version>1.6.2</os.plugin.version>
-        <junit.version>4.13.1</junit.version>
+        <os.plugin.version>1.7.1</os.plugin.version>
```
[the remark this answers is not in the document]
Comment (NihalJain) [reply pom.xml:168]: Sure let me update

**Comment by NihalJain** [comment]
> Please list the changes in the comit message.

Sure let me add details to commit message

**Comment by NihalJain** [comment]
Updated commit message, rebased PR.

**Review by stoty — APPROVED** [review]
+1 LGTM

**Comment by stoty** [comment]
Please check the commit message before commit.
Are the new versions in separate lines ?
It doesn't look like that me.

**Comment by NihalJain** [comment]
> @NihalJain could you please handle the conflicts caused due to other merges.

Rebased code.

**Pull request merged**